modxcms/revolution
# Update Element Panels' Static File/Preview Image UI

### What does it do?

Added new rendering methods for `static_file` and `preview_file` (templates only) fields so that when "None" is selected as the media source, the media browser is disabled and a regular textfield is shown for input.

### Why is it needed?

When a media source is set to "None," the media browser can not account for situations where one may typically use "None" as the source (such as referring to files outside the webroot via an absolute path).

### How to test

1. Ensure your browser cache is cleared and, if you have `compress_js` enabled, rebuild the manager template via `grunt build`.
2. Create one of each element and switch between "None" and your defined media sources to verify the field changes and saving and loading behaves as expected.

### Related issue(s)/PR(s)

Indirectly related to efforts in #16078 and #16070 to shore up the overall behavior of creating and managing static elements.
